imports.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2017 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jsonnet
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+type ImportedData struct {
+	foundHere string
+	content   string
+}
+
+type Importer interface {
+	Import(codeDir string, importedPath string) (*ImportedData, error)
+}
+
+type ImportCacheValue struct {
+	err  error
+	data *ImportedData
+	// nil if we haven't tried to execute it yet
```

Comment:
// nil if we have only imported it via importstr?

Reply:
done